### PR TITLE
fix: Refactor transformer architecture

### PR DESCRIPTION
**Pull request by hego-fdc**
Remove the manual accumulation of commit messages & changes, which makes the transformers easier to understand & debug. It also makes future changes easier to implement.

It slightly changes the commit message for skipped services, making it shorter while still conveing the same information.

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:231]: Why do we need a `[][]string` for the stack instead of just `[]string`?
if we really need it, please add a small comment explaining the dimensions, like
 `// outer array is for foo, inner array is for bar`

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/cd-service/pkg/repository/transformer.go
Comment (hego-fdc) [reply services/cd-service/pkg/repository/transformer.go:235]: Sure, I added an explanation. It's not the most obvious thing ever, but since it is basically invisible when you're writing a transformer, I think it's ok.

**Comment by sven-urbanski-freiheit-com** [comment]
please wait for a review from @hannesg  too

**Comment by hannesg** [comment]
I think you have some room to simplify it even more.

**Pull request merged**
